flex/engines/graph_db/grin/src/predefine.h

```diff
@@ -0,0 +1,66 @@
+#include "grin/predefine.h"
+
+#include "storages/rt_mutable_graph/mutable_property_fragment.h"
+
+typedef gs::oid_t GRIN_OID_T;
+typedef gs::vid_t GRIN_VID_T;
+
+typedef gs::MutablePropertyFragment GRIN_GRAPH_T;
+
+typedef struct GRIN_VERTEX_T {
+  uint8_t label;
+  uint32_t vid;
+} GRIN_VERTEX_T;
+
+typedef struct GRIN_EDGE_T {
+  GRIN_VERTEX_T dst;
+  GRIN_VERTEX_T src;
+  GRIN_DIRECTION dir;
+  gs::label_t label;
+  gs::Any data;
+} GRIN_EDGE_T;
+
+#ifdef GRIN_ENABLE_ADJACENT_LIST
+typedef struct GRIN_ADJACENT_LIST_T {
+  GRIN_VERTEX_T v;
+  GRIN_DIRECTION dir;
+  GRIN_EDGE_TYPE edge_label;
+} GRIN_ADJACENT_LIST_T;
+#endif
+
+#ifdef GRIN_ENABLE_ADJACENT_LIST_ITERATOR
+typedef struct GRIN_ADJACENT_LIST_ITERATOR_T {
+  std::shared_ptr<gs::MutableCsrConstEdgeIterBase> edge_iter;
+  GRIN_ADJACENT_LIST_T* adj_list;
+} GRIN_ADJACENT_LIST_ITERATOR_T;
+#endif
+
+#ifdef GRIN_WITH_VERTEX_PROPERTY
+typedef std::vector<gs::label_t> GRIN_VERTEX_TYPE_LIST_T;
+
+typedef struct GRIN_VERTEX_PROPERTY_T {
+  std::string name;
```

Comment:
[future optimize] caching static schema-data in the graph handle would be a good practice to simplify the data structures for better performance. Please refer to the v6d implementation for caching.